platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/service/WritableRosterStore.java

```diff
@@ -0,0 +1,173 @@
+        roundRosterPairs.addFirst(new RoundRosterPair(round, activeRosterHash));
+        final Builder rosterStateBuilder = RosterState.newBuilder()
+                .candidateRosterHash(previousRosterState.candidateRosterHash())
+                .roundRosterPairs(roundRosterPairs);
+        update(rosterStateBuilder);
+
+        // update the roster map
+        update(ProtoBytes.newBuilder().value(activeRosterHash).build(), roster);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void setCandidateRoster(@NonNull final Roster roster) {
+        Objects.requireNonNull(roster);
+        RosterValidator.validate(roster);
+
+        // update the roster state
+        final RosterState previousRosterState = rosterStateOrThrow();
+        final Bytes candidateRosterHash = RosterUtils.hashOf(roster).getBytes();
+        final Builder rosterStateBuilder = RosterState.newBuilder()
+                .candidateRosterHash(candidateRosterHash)
+                .roundRosterPairs(previousRosterState.roundRosterPairs());
+        update(rosterStateBuilder);
+
+        // update the roster map
+        update(ProtoBytes.newBuilder().value(candidateRosterHash).build(), roster);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void adoptCandidateRoster(final long roundNumber) {
+        final RosterState previousRosterState = rosterStateOrThrow();
+        final Roster candidateRoster = rosterMap.get(ProtoBytes.newBuilder()
+                .value(previousRosterState.candidateRosterHash())
+                .build());
+        if (candidateRoster == null) {
+            throw new IllegalStateException("Candidate roster not found in the state.");
+        }
+        RosterValidator.validate(candidateRoster);
+
+        setActiveRoster(candidateRoster, roundNumber);
+        removeCandidateRoster();
+    }
+
+    /**
+     * remove the candidate roster from the state.
+     */
+    private void removeCandidateRoster() {
+        final RosterState previousRosterState = rosterStateOrThrow();
+        final Bytes candidateRosterHash = previousRosterState.candidateRosterHash();
+
+        final Builder rosterStateBuilder = RosterState.newBuilder()
+                .candidateRosterHash(Bytes.EMPTY)
+                .roundRosterPairs(previousRosterState.roundRosterPairs());
+        update(rosterStateBuilder);
+
+        rosterMap.remove(ProtoBytes.newBuilder().value(candidateRosterHash).build());
+    }
+
+    /**
+     * returns the roster state or throws an exception if the state is null.
+     * @return the roster state
+     * @throws NullPointerException if the roster state is null
+     * @implNote this method is package-private for testing purposes
+     */
+    @NonNull
+    RosterState rosterStateOrThrow() {
+        return requireNonNull(rosterState.get());
+    }
+
+    private void update(@NonNull final RosterState.Builder rosterStateBuilder) {
```

Comment:
There is no reason that there should be 2 methods with the same name. They are doing different things, and should have unique names